calico_kubernetes/calico_kubernetes.py

```diff
@@ -115,19 +96,22 @@ def status(self, namespace, pod_name, docker_id):
                 workload_id=self.docker_id
             )
         except KeyError:
-            logger.error("Container %s doesn't contain any endpoints" % self.docker_id)
+            # Obtain information from Docker Client and validate container state
+            self._validate_container_state(self.docker_id)
+            logger.error("Container %s doesn't contain any endpoints", self.docker_id)
             sys.exit(1)
 
         # Retrieve IPAddress from the attached IPNetworks on the endpoint
         # Since Kubernetes only supports ipv4, we'll only check for ipv4 nets
-        if not endpoint.ipv4_nets :
+        if not endpoint.ipv4_nets:
             logger.error("Exiting. No IPs attached to endpoint %s", self.docker_id)
             sys.exit(1)
         else:
-            ip_net = list(endpoint.ipv4_nets)
-            if len(ip_net) is not 1:
-                logger.warning("There is more than one IPNetwork attached to endpoint %s", self.docker_id)
-            ip = ip_net[0].ip
+            ip_nets = list(endpoint.ipv4_nets)
+            if len(ip_nets) is not 1:
```

Comment:
This should really be:

```python
if len(ip_nets) != 1
```

Using `is` tests identity, not equality. This instance happens to work because Python caches small integers, but you shouldn't rely upon it &ndash; use the equality operators instead: